### PR TITLE
DOCSUP-8660: Table function s3Cluster

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #22012

### CODE REVIEW

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+toc_priority: 55
+toc_title: s3Cluster
+---
+
+# S3Cluster Table Function {#s3Cluster-table-function}
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:6]:
```suggestion
# s3Cluster Table Function {#s3Cluster-table-function}
```

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Allows processing files from [Amazon S3](https://aws.amazon.com/s3/) in parallel from many nodes in a specified cluster. On initiator, it creates a connection to all nodes in the cluster, discloses asterics in S3 file path, and dispatch each file dynamically. On the worker node, it asks the initiator about the next task to process, processes it. This is repeated until the tasks are finished.
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:8]:
```suggestion
Allows processing files from [Amazon S3](https://aws.amazon.com/s3/) in parallel from many nodes in a specified cluster. On initiator, it creates a connection to all nodes in the cluster, discloses asterics in S3 file path, and dispatches each file dynamically. On the worker node, it asks the initiator about the next task to process, and processes it. This is repeated until all tasks are finished.
```

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+**Syntax**
+
+``` sql
+s3Cluster(cluster_name, source, [access_key_id, secret_access_key,] format, structure)
+```
+
+**Arguments**
+
+-   `cluster_name` — Name of a cluster that is used to build a set of addresses and connection parameters to remote and local servers.
+-   `source` — Bucket url with path to file. Supports following wildcards in readonly mode: `*`, `?`, `{abc,def}` and `{N..M}` where `N`, `M` — numbers, `'abc'`, `'def'` — strings. For more information see [here](../../engines/table-engines/integrations/s3.md#wildcards-in-path).
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:19]:
```suggestion
-   `source` — Bucket url with path to file. Supports following wildcards in readonly mode: `*`, `?`, `{abc,def}` and `{N..M}` where `N`, `M` — numbers, `abc`, `def` — strings. For more information see [here](../../engines/table-engines/integrations/s3.md#wildcards-in-path).
```

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+**Returned value**
+
+A table with the specified structure for reading or writing data in the specified file.
+
+**Examples**
+
+Selecting the data from all files in the cluster `cluster_simple`:
+
+``` sql
+SELECT * from s3Cluster('cluster_simple', 'http://minio1:9001/root/data/{clickhouse,database}/*', 'minio', 'minio123', 'CSV', 'name String, value UInt32, polygon Array(Array(Tuple(Float64, Float64)))') ORDER BY (name, value, polygon);
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:32]:
```suggestion
SELECT * FROM s3Cluster('cluster_simple', 'http://minio1:9001/root/data/{clickhouse,database}/*', 'minio', 'minio123', 'CSV', 'name String, value UInt32, polygon Array(Array(Tuple(Float64, Float64)))') ORDER BY (name, value, polygon);
```

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+-   `format` — The [format](../../interfaces/formats.md#formats) of the file.
+-   `structure` — Structure of the table. Format `'column1_name column1_type, column2_name column2_type, ...'`.
+
+**Returned value**
+
+A table with the specified structure for reading or writing data in the specified file.
+
+**Examples**
+
+Selecting the data from all files in the cluster `cluster_simple`:
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:29]:
```suggestion
Select the data from all files in the cluster `cluster_simple`:
```

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+``` sql
+SELECT * from s3Cluster('cluster_simple', 'http://minio1:9001/root/data/{clickhouse,database}/*', 'minio', 'minio123', 'CSV', 'name String, value UInt32, polygon Array(Array(Tuple(Float64, Float64)))') ORDER BY (name, value, polygon);
+```
+
+Count the total amount of rows in all files in the cluster `cluster_simple`:
+
+``` sql
+SELECT count(*) from s3Cluster('cluster_simple', 'http://minio1:9001/root/data/{clickhouse,database}/*', 'minio', 'minio123', 'CSV', 'name String, value UInt32, polygon Array(Array(Tuple(Float64, Float64)))');
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:38]:
```suggestion
SELECT count(*) FROM s3Cluster('cluster_simple', 'http://minio1:9001/root/data/{clickhouse,database}/*', 'minio', 'minio123', 'CSV', 'name String, value UInt32, polygon Array(Array(Tuple(Float64, Float64)))');
```

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+!!! warning "Warning"
+    If your listing of files contains number ranges with leading zeros, use the construction with braces for each digit separately or use `?`.
+
+**See Also**
+
+-   [S3 engine](../../engines/table-engines/integrations/s3.md)
+-   [S3 table function](../../sql-reference/table-functions/s3.md)
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:47]:
```suggestion
-   [s3 table function](../../sql-reference/table-functions/s3.md)
```

Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:47]:
```suggestion
-   [S3 engine](../../engines/table-engines/integrations/s3.md)
-   [s3 table function](../../sql-reference/table-functions/s3.md)
```

Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:47]:
```suggestion
-   [s3 table function](../../sql-reference/table-functions/s3.md)
```

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+**Syntax**
+
+``` sql
+s3Cluster(cluster_name, source, [access_key_id, secret_access_key,] format, structure)
+```
+
+**Arguments**
+
+-   `cluster_name` — Name of a cluster that is used to build a set of addresses and connection parameters to remote and local servers.
+-   `source` — Bucket url with path to file. Supports following wildcards in readonly mode: `*`, `?`, `{abc,def}` and `{N..M}` where `N`, `M` — numbers, `abc`, `def` — strings. For more information see [Wildcards In Path](../../engines/table-engines/integrations/s3.md#wildcards-in-path).
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:19]:
```suggestion
-   `source` — Bucket url with path to file. Supports following wildcards in readonly mode: `*`, `?`, `{'abc','def'}` and `{N..M}` where `N`, `M` — numbers, `abc`, `def` — strings. For more information see [Wildcards In Path](../../engines/table-engines/integrations/s3.md#wildcards-in-path).
```

File: docs/ru/sql-reference/table-functions/s3Cluster.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
+---
+toc_priority: 55
+toc_title: s3Cluster
+---
+
+# Табличная Функция s3Cluster {#s3Cluster-table-function}
```
Comment (olgarev) [inline docs/ru/sql-reference/table-functions/s3Cluster.md:6]:
```suggestion
# Табличная функция s3Cluster {#s3Cluster-table-function}
```

File: docs/ru/sql-reference/table-functions/s3Cluster.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+Позволяет обрабатывать файлы из [Amazon S3](https://aws.amazon.com/s3/) параллельно из многих узлов в указанном кластере. На узле-инициаторе функция создает соединение со всеми узлами в кластере, раскрывает звездочки в пути к файлу S3 и динамически отправляет каждый файл. На рабочем узле функция запрашивает у инициатора следующую задачу для обработки и обрабатывает ее. Это повторяется до тех пор, пока все задачи не будут завершены.
```
Comment (olgarev) [inline docs/ru/sql-reference/table-functions/s3Cluster.md:8]:
```suggestion
Позволяет обрабатывать файлы из [Amazon S3](https://aws.amazon.com/s3/) параллельно из многих узлов в указанном кластере. На узле-инициаторе функция создает соединение со всеми узлами в кластере, раскрывает символы '*' в пути к файлу S3 и динамически отправляет каждый файл. На рабочем узле функция запрашивает у инициатора следующую задачу и обрабатывает ее. Это повторяется до тех пор, пока все задачи не будут завершены.
```

File: docs/ru/sql-reference/table-functions/s3Cluster.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+**Синтаксис**
+
+``` sql
+s3Cluster(cluster_name, source, [access_key_id, secret_access_key,] format, structure)
+```
+
+**Аргументы**
+
+-   `cluster_name` — имя кластера, используемое для создания набора адресов и параметров подключения к удаленным и локальным серверам.
+-   `source` — URL-адрес бакета с указанием пути к файлу. Поддерживает следующие подстановочные символы в режиме "только чтение": `*`, `?`, `{abc,def}` и `{N..M}`, где `N`, `M` — числа, `abc`, `def` — строки. Подробнее смотрите в разделе [Символы подстановки](../../engines/table-engines/integrations/s3.md#wildcards-in-path).
```
Comment (olgarev) [inline docs/ru/sql-reference/table-functions/s3Cluster.md:19]:
```suggestion
-   `source` — URL-адрес бакета с указанием пути к файлу. Поддерживает следующие подстановочные символы в режиме "только чтение": `*`, `?`, `{'abc','def'}` и `{N..M}`, где `N`, `M` — числа, `abc`, `def` — строки. Подробнее смотрите в разделе [Символы подстановки](../../engines/table-engines/integrations/s3.md#wildcards-in-path).
```

File: docs/ru/sql-reference/table-functions/s3Cluster.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+**Примеры**
+
+Выведем данные из всех файлов кластера `cluster_simple`:
+
+``` sql
+SELECT * FROM s3Cluster('cluster_simple', 'http://minio1:9001/root/data/{clickhouse,database}/*', 'minio', 'minio123', 'CSV', 'name String, value UInt32, polygon Array(Array(Tuple(Float64, Float64)))') ORDER BY (name, value, polygon);
+```
+
+Подсчитаем общее количество строк во всех файлах кластера `cluster_simple`:
```
Comment (olgarev) [inline docs/ru/sql-reference/table-functions/s3Cluster.md:36]:
```suggestion
Подсчет общего количества строк во всех файлах кластера `cluster_simple`:
```

File: docs/ru/sql-reference/table-functions/s3Cluster.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+-   `format` — [формат](../../interfaces/formats.md#formats) файла.
+-   `structure` — структура таблицы. Формат `'column1_name column1_type, column2_name column2_type, ...'`.
+
+**Возвращаемое значение**
+
+Таблица с указанной структурой для чтения или записи данных в указанный файл.
+
+**Примеры**
+
+Выведем данные из всех файлов кластера `cluster_simple`:
```
Comment (olgarev) [inline docs/ru/sql-reference/table-functions/s3Cluster.md:30]:
```suggestion
Вывод данных из всех файлов кластера `cluster_simple`:
```

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+**Syntax**
+
+``` sql
+s3Cluster(cluster_name, source, [access_key_id, secret_access_key,] format, structure)
+```
+
+**Arguments**
+
+-   `cluster_name` — Name of a cluster that is used to build a set of addresses and connection parameters to remote and local servers.
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/s3Cluster.md:18]:
```suggestion
-   `cluster_name` — Name of a cluster that is used to build a set of addresses and connection parameters to remote and local servers. [String](../../sql-reference/data-types/string.md).
```

**Comment by sevirov** [comment]
@nikitamikhaylov Please, review this PR.

File: docs/en/sql-reference/table-functions/s3Cluster.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+-   `source` — Bucket url with path to file. Supports following wildcards in readonly mode: `*`, `?`, `{'abc','def'}` and `{N..M}` where `N`, `M` — numbers, `abc`, `def` — strings. For more information see [Wildcards In Path](../../engines/table-engines/integrations/s3.md#wildcards-in-path).
```
Comment (nikitamikhaylov) [inline docs/en/sql-reference/table-functions/s3Cluster.md:19]:
```suggestion
-   `source` — URL to a file or a bunch of files. Supports following wildcards in readonly mode: `*`, `?`, `{'abc','def'}` and `{N..M}` where `N`, `M` — numbers, `abc`, `def` — strings. For more information see [Wildcards In Path](../../engines/table-engines/integrations/s3.md#wildcards-in-path).
```

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**